extensions/wikia/AdEngine/js/AdEngine2.js

```diff
@@ -68,11 +80,17 @@ define('ext.wikia.adEngine.adEngine', [
 					provider = providerList.shift();
 
 					if (!provider) {
-						return fillInSlotUsingProvider(slotname, adProviderNull, noop);
+						delete slot.success;
+
+						if (slot.error) {
```

Comment:
Would typeof slot.error === 'function' make sense here or it's sure to be a function when it's set?